js/wavesexchange.js

```diff
@@ -1159,14 +1168,17 @@ module.exports = class wavesexchange extends Exchange {
     }
 
     currencyFromPrecision (currency, amount) {
-        const scale = this.currencies[currency]['precision'];
+        const scale = this.precisionFromString (this.safeString (this.currencies[currency], 'precision'));
         return this.fromPrecision (amount, scale);
     }
 
     priceFromPrecision (symbol, price) {
         const market = this.markets[symbol];
-        const wavesPrecision = this.safeInteger (this.options, 'wavesPrecision', 8);
-        const scale = wavesPrecision - market['precision']['amount'] + market['precision']['price'];
+        const precision = market['precision'];
+        const precisionAmountDigits = this.precisionFromString (this.safeString (precision, 'amount'));
+        const precisionPriceDigits = this.precisionFromString (this.safeString (precision, 'price'));
+        const wavesPrecision = this.safeString (this.options, 'wavesPrecision', '0.00000001');
+        const scale = this.precisionFromString (wavesPrecision) - precisionAmountDigits + precisionPriceDigits;
```

Comment:
Using the operator `+` here is a violation of this rule:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings

The bug was there before this PR, however.

Reply:
yes, I see, but if possible, I'd lean also to concentrate on specific line of issues in these series of PRs, so, for the sake to speed up merging, I'll not touch other issues, and after the Tick-size project is complete, I'll take another task to correct all those violations (separate project named "arithmetic 
 corrections" or alike)